hot-reload components referencing k8s secrets (#5234)
dapr/docs: daprdocs/content/en/developing-applications/building-blocks/secrets/secrets-overview.md

When configuring Dapr components such as state stores, you're often required to include credentials in components files. Alternatively, you can place the credentials within a Dapr supported secret store and reference the secret within the Dapr component. This is the preferred approach and recommended best practice, especially in production environments.

> When running in Kubernetes and referencing secrets from the built-in Kubernetes secret store, Dapr also automatically reloads components when the referenced secrets change, for example when a credential is rotated, without restarting the application.

Reviewer: Add somewhere that this only works when the hot-reload functionality is enabled

> When running in Kubernetes and referencing secrets from the built-in Kubernetes secret store, Dapr also automatically reloads components when the referenced secrets change, for example when a credential is rotated, without restarting the application. This requires [hot reloading]({{% ref "component-updates.md#hot-reloading" %}}) to be enabled, which is the default.

Reviewer 2: ```suggestion
When running in Kubernetes and referencing secrets from the built-in Kubernetes secret store, Dapr also automatically reloads components when the referenced secret values change, for example when a credential is rotated, without restarting the application. This is enabled by default using the [hot reloading]({{% ref "component-updates.md#hot-reloading" %}}) feature.
```
